src/boot/efi/boot.c

```diff
@@ -2251,6 +2291,8 @@ static EFI_STATUS initrd_prepare(const ConfigEntry *entry, CHAR16 **ret_options)
         }
 
         *ret_options = TAKE_PTR(options);
+        *ret_initrd = TAKE_PTR(initrd);
+        *ret_initrd_size = size;
```

Comment:
the upper EFI_SUCESS code paths should initialize these to NULL/0, as per coding style